autotest.pm

```diff
@@ -393,12 +395,15 @@ sub runalltests {
                 return 0;
             }
             elsif (!$flags->{no_rollback} && $last_milestone) {
+                $current_test->record_resultfile('loading snapshot', 'milestone requested');
                 load_snapshot('lastgood');
                 $last_milestone->rollback_activated_consoles();
             }
         }
         else {
             if (!$flags->{no_rollback} && $last_milestone && $flags->{always_rollback}) {
+                die "ERROR: no tests loaded" unless $current_test;
```

Comment:
did you ever encounter this case? I don't see you adding a test case for this so I wonder under what circumstances this would be needed. Please also use single tick strings by default

Reply:
It looks like the case where one would use the `assert`-macro in C. Likely we should trust our understanding of the code that this case can never happen or we at least use a better message with a bit more context (e.g. "Unable to roll back as no tests have been loaded.") and add a test.

Reply:
Right. I wouldn't have worried if the complete module has been introduced with this. I mostly wonder why the author introduced this line :)

Reply:
And I tried to inestigate it but honestly I don't get how we can be loading a snapshot when there's no test here

Reply:
Sure, trying to load a snapshot when there is no test won't work. The question was how you actually encountered the case of no test being  defined here. You stated you found out by "running the tests". I don't understand what you mean. You haven't added tests for this and previously CI tests have passed, so which tests do you mean? Manual tests?

Reply:
The unit tests don't pass because they die due to the undefined value, even without the explicit `die` message. You can see this in CI.

Reply:
I tried out myself and progressed a bit but haven't succeeded completely with the `record_resultfile` approach. This is my diff: